### PR TITLE
Changed Dockerfile to work with react framework

### PR DESCRIPTION
**Pull request by maxb2**
Fixes apclary/dungeon-revealer#50.

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -1,19 +1,27 @@
...
 # Install app dependencies
 # A wildcard is used to ensure both package.json AND package-lock.json are copied
 # where available (npm@5+)
-COPY package*.json ./
+#COPY package*.json ./
+COPY . .
+
 
+RUN npm set unsafe-perm true
+RUN npm i eslint-plugin-import eslint-plugin-flowtype
```
Comment (maxb2) [inline Dockerfile]: It would be better if these dependencies were in package.json.

Comment (apclary) [reply Dockerfile]: I'm a little confused, why do you need to install additional dependencies? Isn't eslint in the node_modules/.bin/ folder? I was able to run eslint by running `npm run eslint` IIRC.

Comment (n1ru4l) [reply Dockerfile]: I have to agree with @apclary, these additional dependencies shouldn't be required.

**Comment by n1ru4l** [comment]
We could also add an additional CI task that checks whether the image is working or not what do you think?

File: Dockerfile (hunk 1 above)
Comment (apclary) [reply Dockerfile]: Also minor thing but we could use a higher docker version, I used 10 for the executables but open-minded if we should use something even higher.

Comment (maxb2) [reply Dockerfile]: You're right. I had to do that previously when I was copying in the app source via `COPY package*.json`. Using `COPY . .` makes the separate install step unnecessary.

**Comment by apclary** [comment]
I'm concerned that github doesn't see the dockercloud build status. I think I need to configure something so it will automatically build pull requests.

**Comment by n1ru4l** [comment]
@apclary Do we really need a dockercloud build for each pull requests? I think it would be enough to try building the image on CI (and do a smoke test by spinning it up to curl the front page etc.). On the other hand, having a dockercloud image for each pull request would allow easier pull request/feature reviews/previews (e.g. user has to only run `docker run xyz@foobars`).

**Comment by apclary** [comment]
@n1ru4l that makes sense, I will remove the dockercloud status.

**Pull request merged**